go/testimonyd/internal/socket/socket.go

```diff
@@ -120,13 +122,12 @@ func (s *socket) String() string {
 // which the run() method passes to clients.
 func (s *socket) getNewBlocks() {
```

Comment:
switch this function to return an error, and pass it up the chain?

Reply:
Added error return value and now it's handled inside goroutine.